### PR TITLE
Add ``_ArgumentsManager.set_option`` and deprecate old option setters

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.
- [x] If you used multiple emails or multiple names when contributing, add your mails
   and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |

## Description

Bit of a larger one.

@Pierre-Sassoulas  Let me know what you think of my hack of deprecating unwanted arguments. It's not pretty, but it might work and is more efficient than doing it with `inspect` or something like that.

### CODE REVIEW

**Comment by DanielNoord** [comment]
Coverage drop is as expected.

File: pylint/checkers/variables.py [hunk 1]
```diff
@@ -988,7 +988,7 @@ class VariablesChecker(BaseChecker):
         (
             "init-import",
             {
-                "default": 0,
+                "default": False,
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/variables.py:991]: 👍

File: pylint/checkers/similar.py [hunk 2]
```diff
@@ -798,33 +798,18 @@ def set_option(self, optname, value, action=None, optdict=None):
...
-        BaseChecker.set_option(self, optname, value, action, optdict)
+        # TODO: Deprecate the non required arguments
+        self.linter.set_option(optname, value)
         if optname == "min-similarity-lines":
-            self.min_lines = (
-                getattr(self.linter.namespace, "min_similarity_lines", None)
-                or self.config.min_similarity_lines
-            )
+            self.min_lines = self.linter.namespace.min_similarity_lines
         elif optname == "ignore-comments":
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/similar.py:805]: Do we need the conditional and set_option here ? Can't we do 
```python
self.min_lines = self.linter.namespace.min_similarity_lines
...
self.ignore_signatures = self.linter.namespace.ignore_signatures
```
Directly inside the constructor of ``Similar`` ?

Comment (DanielNoord) [reply pylint/checkers/similar.py:805]: We probably can, but I need to refactor this anyway. (See the TODO).

I'll take a look at these assignments at the same time. I'm likely going to deprecate this complete method and just use `set_option` from the `Manager`. No need to duplicate if both work fine.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**